support/crown/tests/compile_test.rs

```diff
@@ -0,0 +1,42 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use std::env;
+use std::path::PathBuf;
+
+use compiletest_rs as compiletest;
+use once_cell::sync::Lazy;
+
+static PROFILE_PATH: Lazy<PathBuf> = Lazy::new(|| {
+    let current_exe_path = env::current_exe().unwrap();
+    let deps_path = current_exe_path.parent().unwrap();
+    let profile_path = deps_path.parent().unwrap();
+    profile_path.into()
+});
+
+fn run_ui_tests(bless: bool) {
+    let mut config = compiletest::Config {
+        bless,
+        edition: Some("2021".into()),
+        mode: compiletest::common::Mode::Ui,
+        ..Default::default()
+    };
+
+    config.target_rustcflags = Some(format!(
+        "-Zcrate-attr=feature(register_tool) -Zcrate-attr=register_tool(crown)"
+    ));
+
+    config.src_base = "tests/ui".into();
+    config.build_base = PROFILE_PATH.join("test/ui");
+    config.rustc_path = PROFILE_PATH.join("crown");
+    config.link_deps(); // Populate config.target_rustcflags with dependencies on the path
+
+    compiletest::run_tests(&config);
+}
```

Comment:
I'm not sure this function is buying us anything. I would merge it into `fn compile_test` below.